src/cargo/core/compiler/custom_build.rs

```diff
@@ -916,10 +916,8 @@ impl BuildOutput {
                     if extra_check_cfg {
                         check_cfgs.push(value.to_string());
                     } else {
-                        warnings.push(format!(
-                            "{}{} requires -Zcheck-cfg flag",
-                            syntax_prefix, key
-                        ));
+                        // silently ignoring the instruction to try to
+                        // minimise MSRV annoyance when stabilizing -Zcheck-cfg
                     }
                 }
```

Comment:
wouldn't this create the same issue we trying to avoid?

or you mean that the silencing part should be nightly/stable conditioned ?

Reply:
Do you mean that it should not be silenced on nightly, so that nightly users get the warning (which is useful!)?

Reply:
The main problem we are trying to avoid is when this is stabilized but not yet in people's MSRV: they will either have to live with the build script warning on their MSRV or deal with the check-cfg warning on the latest Rust toolchain.

(btw I keep going back and forth between "meh, this is fine" and "we probably should still warn")